### PR TITLE
Fix: handle error with uploaded logo on carrier form

### LINKED ISSUE #36759: New page carriers - bugs
### Prerequisites

- [X] I understand and accept the project's [code of conduct](https://github.com/PrestaShop/PrestaShop/blob/develop/CODE_OF_CONDUCT.md).
- [X] I have already [searched in existing issues](https://github.com/PrestaShop/PrestaShop/issues?q=is%3Aissue+label%3ABug) and found no previous report of this bug.

### Describe the bug and add attachments

Bug 1 
I edit an old carrier, if I want to add a new zone but I do not enter the cost, the name of the zone disappears.

Bug 2
I create a new carrier, if I want to add a new zone, I cannot enter a cost.

Bug3
If I want to put a logo greater than 8 Mb, I have an error 500

Bug 4
If I configure my carrier with maximum authorized order dimensions of 1cm x 1cm x 1cm. I can order an order greater than these dimensions if the weight of the order is less than the maximum authorized weight.

Bug 5 
If the shipping costs are free, I cannot choose a delivery zone.

Bug 6
I can't save a carrier if all the required fields are not filled in. If I don't fill them in, I don't get an error message to tell me.

### Expected behavior

_No response_

### Steps to reproduce

Bug 1 :
https://www.loom.com/share/c1ba7893e5174e9591dd4397f4f2d1f7

Bug 2 : 
https://www.loom.com/share/d42a86ca6d074792a49035b8d3a034db

Bug 3 : 
https://www.loom.com/share/a3f932975bbb4a1db1e712d5f3eaeac2

Bug 4 : 
image
image

Bug 5 : 
image

Bug 6
https://www.loom.com/share/581a1052cdc14effb8a4a8f3a5582d68

### PrestaShop version(s) where the bug happened

develop

### PHP version(s) where the bug happened

8.1

### If your bug is related to a module, specify its name and its version

_No response_

### Your company or customer's name goes here (if applicable).

_No response_

### PR DESCRIPTION
**Pull request by Poulinhoo**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | Handle & display errors with logo upload on new carrier page
| Type?             | bug fix
| Category?         | BO
| BC breaks?        |  no
| Deprecations?     | no
| How to test?      | Follow steps on issue [#36759](https://github.com/PrestaShop/PrestaShop/issues/36759) number 3
| Fixed issue or discussion?     | Fixes [#36759](https://github.com/PrestaShop/PrestaShop/issues/36759)
| UI test | https://github.com/PoulainMaxime/ga.tests.ui.pr/actions/runs/10828500149

### CODE REVIEW

**Comment by ps-jarvis** [comment]
This pull request seems to contain new translation strings. I have summarized them below to ease up review:

- `Admin.Carriers.Form.LogoUpload` <span title="New domain!">⚠️</span>
    - [ ] `Please upload a valid jpeg file`
    - [ ] `The file is too large. Allowed maximum size is 8MB.`

(Note: this is an automated message, but answering it will reach a real human)

**Comment by kpodemski** [comment]
I think that instead of hardcoded value you could use:

`$uploadMaxSize = (int) str_replace('M', '', ini_get('upload_max_filesize'));`, same as in `src/Adapter/Upload/UploadQuotaConfiguration.php`

**Review by jolelievre — DISMISSED** [review]
Thanks @PoulainMaxime

File: src/PrestaShopBundle/Form/Admin/Improve/Shipping/Carrier/GeneralSettings.php [hunk 1]
```diff
@@ -52,6 +53,8 @@ public function __construct(
 
     public function buildForm(FormBuilderInterface $builder, array $options)
     {
+        $maximumFileSize = (int) str_replace('M', '', ini_get('upload_max_filesize'));
```
Comment (jolelievre) [inline src/PrestaShopBundle/Form/Admin/Improve/Shipping/Carrier/GeneralSettings.php]: Actually to match with the CQRS implementation you should use `PrestaShop\PrestaShop\Adapter\Carrier\Validate\CarrierValidator::MAX_IMAGE_SIZE_IN_BYTES`

**Review by tleon — COMMENTED** [review]
Can you run the UI test on this PR pls ?

**Comment by Poulinhoo** [comment]
> Can you run the UI test on this PR pls ?

I update the description with the UI tests, everything is ok

**Review by jolelievre — APPROVED** [review]

**Review by boherm — APPROVED** [review]

**Review by matthieu-rolland — APPROVED** [review]

**Comment by ps-jarvis** [comment]
PR merged, well done!

Message to @PrestaShop/committers: do not forget to milestone it before the merge.

**Pull request merged**